Source/Cells/ImageDownloader.swift

```diff
@@ -9,13 +9,44 @@
 import UIKit
 
 // Mark: - Image Downloader
-public protocol ImageDownloader {
+public protocol ImageDownloader: class {
     func downloadImage(with imageURL: NSURL, onCompletion completionHandler: ((image: UIImage, url: NSURL) -> Void))
+    func downloadImageAndSet(on imageView: UIImageView, with imageURL: NSURL, onCompletion completionHandler: ((image: UIImage, url: NSURL) -> Void))
 }
 
-class NSURLSessionImageDownloader: ImageDownloader {
+public extension ImageDownloader {
 
-    func downloadImage(with imageURL: NSURL, onCompletion completionHandler: ((image: UIImage, url: NSURL) -> Void)) {
+    func downloadImageAndSet(on imageView: UIImageView, with imageURL: NSURL, onCompletion completionHandler: ((image: UIImage, url: NSURL) -> Void)) {
+        self.downloadImage(with: imageURL) { (image, url) in
+            guard imageURL == url else {
+                return
+            }
+
+            NSOperationQueue.mainQueue().addOperationWithBlock {
+                imageView.image = image
+                completionHandler(image: image, url: url)
+            }
+        }
+    }
+
+}
+
+public class NSURLSessionImageDownloader: ImageDownloader {
+
+    public func downloadImageAndSet(on imageView: UIImageView, with imageURL: NSURL, onCompletion completionHandler: ((image: UIImage, url: NSURL) -> Void)) {
```

Comment:
Oh right, I ran into this with Nick LoBue doing something else. I suspect you could defer to the protocol's implementation if you do this within the class's implementation:

```swift
(self as ImageDownloader).downloadImageAndSet(on imageView:...)
```

but it would need testing to confirm that this works as expected.